File: backendmealplan/src/main/java/org/backendmealplan/backendmealplan/beans/GroceryList.java
package org.backendmealplan.backendmealplan.beans;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import javax.persistence.*;
import javax.validation.constraints.NotNull;
import java.util.List;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
@Table(name="grocery_list")
public class GroceryList {


    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long groceryId;

    @NotNull
    private Integer week;
    @NotNull
    private Integer amount;
    private String unit;
    @ManyToOne
    @JoinColumn(name = "ingredient_id")
    private Ingredient ingredient;

    @ManyToOne
    @JoinColumn(name = "plan_id")
    private Plan plan;

    @ManyToMany(mappedBy = "changes")
    List<User> users;

}

File: frontendmealplan/src/app/mealplan/models/GroceryList.ts
import { Ingredient } from "./Ingredient";
import { Plan } from "./Plan";
import { User } from "./User";

//remove fields : plan and users
export class GroceryList {
  constructor(
    public groceryId?: number,
    public week?: number,
    public amount?: number,
    public unit?: string,
    public ingredient?: Ingredient,
    public plan?: Plan,
    public users?: User[]
  ) {}
}
